apps/dashboard/src/components/layouts/layout-editor-provider.tsx

```diff
@@ -54,9 +149,13 @@ export const LayoutEditorProvider = ({
   const [previewContextValue, setPreviewContextValue] = useState('{}');
   const previewContextValueRef = useDataRef(previewContextValue);
   const location = useLocation();
-
+  const { data: organizationSettings, isLoading: isOrgSettingsLoading } = useFetchOrganizationSettings();
```

Comment:
the general improvements here are: 
- preview context local storage persistance and sync between locale dropdown
- moved form components and submit logic